app/models/repository.rb

```diff
@@ -29,12 +29,17 @@
 
 class Repository < ActiveRecord::Base
   include Redmine::Ciphering
+  include OpenProject::Scm::ManageableRepository
 
   belongs_to :project
   has_many :changesets, order: "#{Changeset.table_name}.committed_on DESC, #{Changeset.table_name}.id DESC"
 
   before_save :sanitize_urls
 
+  # Managed repository lifetime
+  after_save :create_managed_repository, if: Proc.new { |repo| repo.managed? }
+  after_destroy :delete_managed_repository, if: Proc.new { |repo| repo.managed? }
```

Comment:
Any chance we can still change the pattern here from the after_safe/after_destroy to notifications? This would enable us to separate the handling of creating/deleting on FS even further from this ActiveRecord.

So the repository would simply throw a "repository_created/destroyed" notification for which some listener is registered. The `Repository` class would then no longer have to know that something like managing exists. The listener would have the complete logic of deciding if a repository has to be created on the FS.